tdp_core/tests/test_rdkit_img.py

```diff
@@ -0,0 +1,82 @@
+from hashlib import sha3_256
+from urllib.parse import quote
+
+import pytest
+
+mol_expected = {
+    "C": "6d435ada2a36f00980347aef78f309de71544180a293f5970d7e257dd88f9270",
+    "O": "0db87f49c83caba8c7f7ca36f3128f929590537d92678ced396afb4c6201ba2d",
+    "OO": "fc1504931d7fde7530be739c2d4cc5083a2a77921a9b6e02df86a11761336d17",
+    "[He]": "502933d4a899062be2c1aae6d14552e79a0baa8e5af349d286fdf0a8deb3a4e8",
+}
+
+similarity_data = [
+    ["CC", "CCCC", "0060b251631425116dac5a98b2db41b69247fe938141cd8e8204f74af2c8ee99"],
+    ["CCCC", "CC", "0c4e2d35451f1aea4c660b38e6e1cddb6ed2361f2ff35efda24bd472244e08c7"],
+]
+
+
+def hash_compare(svg, expected):
+    assert sha3_256(svg).hexdigest() == expected
+
+
+@pytest.mark.parametrize("s", ["H", "He"])
+def test_invalid(client, s):
+    """A single H atom isn't a valid molecule"""
+    res = client.get(f"/api/image/{s}")
+    assert res.status_code == 422
+
+
+@pytest.mark.parametrize("mol, expected", mol_expected.items())
+def test_valid(client, mol, expected):
+    res = client.get(f"/api/image/{mol}")
+    assert res.status_code == 200
+    assert res.headers.get("content-type").startswith("image/svg")
+    hash_compare(res.content, expected)
```

Comment:
Do we really want to hash-compare the SVGs? What if a new version of RDKit just _slightly_ changes a color-coding, rotation, etc.? I know that this is meant as "snapshot" test here, but I think it is sufficient to check if a `<svg` including a `<line` is returned, everything else is probably too much. Image similarity test w/ threshold would be interesting though.

Reply:
in many rdkit-error-cases, it returns an SVG with a big ? in it (technically still an `<svg` with a `<line`) and i'd rather take the risk of having to change hashes once than not detecting that after upgrading all of our images are question marks.

> Image similarity test w/ threshold would be interesting though.
that sounds like 10x complexity